gui/src/pages/NewProjectConfig.svelte

```diff
@@ -0,0 +1,165 @@
+<script lang="ts">
+  import Layout from '../components/Layout.svelte';
+  import Textbox from '../components/Textbox.svelte';
+  import Spinner from '../components/Spinner.svelte';
+  import ErrorLabel from '../components/ErrorLabel.svelte';
+  import SubmitButton from '../components/form/SubmitButton.svelte';
+  import CenterFormPanel from '../components/form/CenterFormPanel.svelte';
+  import type { ReadDirResult } from '../lib/api';
+  import { api, isClientError } from '../lib/api';
+  import * as router from 'svelte-spa-router';
+
+  export let params = { starter: '' };
+
+  const { starter } = params;
+
+  let name = starter;
+  let templateUrl: string | null = null;
+  let error: Error | null = null;
+
+  let workingDirectory: string | null = null;
+
+  const setWorkingDirectory = (dir: string) => {
+    workingDirectory = dir;
+    dirPromise = api.kernel.readDir(workingDirectory);
+  };
+
+  let dirPromise: Promise<ReadDirResult> = api.kernel
+    .getUserHomeDir()
+    .then((dir) => {
+      setWorkingDirectory(dir);
+      return dirPromise;
+    });
+
+  const withSlash = (wd: string) => (wd.slice(-1) === '/' ? wd : wd + '/');
+</script>
+
+<Layout>
+  <CenterFormPanel title={`New ${starter} project`} backRoute="#/new-project">
+    <form
+      on:submit|preventDefault={async () => {
+        error = null;
+        let workspaceId;
+
+        if (starter !== 'empty') {
+          templateUrl = String(
+            (await api.kernel.describeTemplates()).find(
+              (x) => x.name === starter,
+            )?.url,
+          );
+        }
+
+        try {
+          workspaceId = await api.kernel.createProject(
+            `${withSlash(String(workingDirectory))}${name}`,
+            templateUrl,
+          );
+          router.push(`/workspaces/${encodeURIComponent(workspaceId)}`);
+        } catch (ex) {
+          if (!isClientError(ex)) {
```

Comment:
This is a type error in Typescript 4.4. The variable `ex` will be of type `unknown` and `isClientError` takes an `Error`. You need to either cast to `any` or check if it's an error. 

https://devblogs.microsoft.com/typescript/announcing-typescript-4-4/#use-unknown-catch-variables

Reply:
Does this fix work? https://github.com/deref/exo/pull/337/commits/a4c80a1c508109e37cb4c72227d298cb53fb9d4b#

Reply:
Not really — now it ignores the error if it isn't of type `Error`. What we really want to do is rethrow the error if it isn't the correct type.

Something like
```
if (!(ex instanceof Error) || !isClientError(ex)) {
throw ex;
}
error = ex;
```

Reply:
Okay done in 0f1e099